add_book.php

```diff
@@ -0,0 +1,34 @@
+<?php
+include 'database.php';
+
+if ($_SERVER['REQUEST_METHOD'] == 'POST') {
```

Comment:
This code does not currently have any unit tests. Consider adding tests to ensure that the logic for adding a book is functioning correctly. You can use PHPUnit for this purpose.

Example test:

```php
use PHPUnit\Framework\TestCase;

class BookRepositoryTest extends TestCase {
    public function testAddBook() {
        $mockConnection = $this->createMock(mysqli::class);
        $mockStmt = $this->createMock(mysqli_stmt::class);

        $mockConnection->method('prepare')->willReturn($mockStmt);
        $mockStmt->expects($this->once())->method('bind_param')->with($this->equalTo('ssi'), $this->anything(), $this->anything(), $this->anything());
        $mockStmt->method('execute')->willReturn(true);

        $repository = new BookRepository($mockConnection);
        $this->assertTrue($repository->addBook('Test Title', 'Test Author', 2021));
    }
}
```